### PR TITLE
Feature: Tuition App Revision

### PR DESCRIPTION
**Pull request by Andriy-Kulak**
Work Reference: https://www.aikido.work/bounty/xE4QIDGA28I8BlCoQar4YcHw

UI for this PR can be tested here: https://tuition-dapp-pr-3.onrender.com/

**Done:**
 - update component to used ETH pegged to USD
 - updated text to reflect change
 - removed unnecessary react state & cleaned up code
 - added `Getting Started` & note in Readme since documentation is sparse
- ensured that we are hardcoding eth to be sent to 0xmacro.eth
- removed utils that are no longer necessary to interact with contract. If I didn't remove them, then we would have to run `yarn hardhat compile` to make front-end work locally even though we weren't using the contract at all
- enabled preview deployments on render for this repo

**Tested the following flow on rinkeby:**

✅ Insufficient funds pops up correctly
✅ when I switch metamask accounts, Connected wallet is updated in web app
✅ loading and success messages seems to work 
✅ ETH / USD pegging price works well
✅ when metamask is not connected, "Connect Wallet" is displayed and forces user to log in
✅ tested sending $1 USD in real eth using flow: https://etherscan.io/tx/0xaa17c4b0f2bc86fe2fef8ca1ec53f4f48139d08386160940627d755e3497810f

**Things I did not test:**
- non-metamask wallets
- didn't test sending real 1.67 eth to macro treasury ;-)

**Peg seems to work well. Most I saw the estimate off metamask be is $5.**
Screen Shot 2022-08-08 at 19 01 14

**Same thank you message afterwards**
Screen Shot 2022-08-08 at 19 17 11

**Future considerations:**
- we may remove token contract & any relevant references completely from repo

### CODE REVIEW

File: utils/constants.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// production wallet for 0xmacro.eth
+export const TREASURY_ADDRESS = "0x705a47ebc6fce487a3c64a2da64ce2e3b8b2ef55";
```
Comment (Andriy-Kulak) [inline utils/constants.ts:2]: Please triple check and confirm that this is the correct treasury address

**Comment by dd0sxx** [comment]
Please add a button which allows students the option to pay 3,000 USDC, in addition to the button you added which allows them to send an equivalent amount of ETH 👍

Will triple check the treasury address and the rest of the code as well now ^_^

File: utils/constants.ts (hunk 1 above)
Comment (dd0sxx) [reply utils/constants.ts:2]: This is correct 💯

**Comment by Andriy-Kulak** [comment]
> Please add a button which allows students the option to pay 3,000 USDC, in addition to the button you added which allows them to send an equivalent amount of ETH 👍
> 
> Will triple check the treasury address and the rest of the code as well now ^_^

added button! everything should work more or less the same. I tested sending 0.1 usdc and 0.001 eth via this flow. References


✅ **Sent 0.1 USDC:** https://etherscan.io/tx/0xcfbcfe804ffbc63c1dab3ae6c07329a1d2659b0a21f114c4e2d14ad8779cba56
✅ **sent tiny eth:** https://etherscan.io/tx/0x348032f8e1d3e25ebc1b3c0462443f6ae18e2b1254cf52f2f1b020b8cedd99f2

✅ added addition state and error considerations with USDC.
✅ updated text per our conversation

File: utils/constants.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+// USDC token info. reference: https://etherscan.io/token/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48
+export const USDC_ERC20_ADDRESS = "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48";
+export const USDC_DECIMALS = 6;
```
Comment (Andriy-Kulak) [inline utils/constants.ts:6]: Please double check that this is the right USDC address and decimal count

**Comment by dd0sxx** [comment]
Please add some logic to ensure the user is on the right chain (mainnet ethereum)

**Comment by Andriy-Kulak** [comment]
> Please add some logic to ensure the user is on the right chain (mainnet ethereum)

done in this commit. https://github.com/0xMacro/tuition/pull/3/commits/d3e4024d491ad603651b7604798b2740ec534dc8

here is the message if you try a test net
Screen Shot 2022-08-10 at 12 59 41

**Review by dd0sxx — APPROVED** [review]
Looks good!

File: utils/constants.ts (hunk 2 above)
Comment (dd0sxx) [reply utils/constants.ts:6]: lgtm 👍

**Pull request merged**